src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Microsoft.Win32.SafeHandles;
+using System.Diagnostics;
+
+namespace System.Net.Sockets
+{
+#if DEBUG
+    internal sealed partial class SafeCloseSocket : DebugSafeHandleMinusOneIsInvalid
+#else
+    internal sealed partial class SafeCloseSocket : SafeHandleMinusOneIsInvalid
+#endif
```

Comment:
To further highlight the debug difference, this could be:
```C#
internal sealed partial class SafeCloseSocket :
#if DEBUG
    DebugSafeHandleMinusOneIsInvalid
#else
    SafeHandleMinusOneIsInvalid
#endif
```